freqtrade/optimize/__init__.py

```diff
@@ -134,23 +180,33 @@ def download_backtesting_testdata(datadir: str, pair: str, interval: str = '5m')
         interval=interval,
     ))
 
+    since = get_start_ts_from_timerange(timerange, interval)
+
     if os.path.isfile(filename):
         with open(filename, "rt") as file:
             data = json.load(file)
-        logger.debug("Current Start: %s", misc.format_ms_time(data[1][0]))
-        logger.debug("Current End: %s", misc.format_ms_time(data[-1:][0][0]))
+
+        if since:
+            if since < data[0][0]:
+                # fully update the data
+                data = []
+            else:
+                # download unexist data only
+                since = max(since, data[-1][0] + 1)
+        else:
+            # download unexist data only
+            since = data[-1][0] + 1
     else:
         data = []
-        logger.debug("Current Start: None")
-        logger.debug("Current End: None")
 
-    new_data = get_ticker_history(pair=pair, tick_interval=interval)
-    for row in new_data:
-        if row not in data:
-            data.append(row)
```

Comment:
If there is the existing data `since` parameter will be equal to the `max({requested time by user}, {last exist data time} + 1)` then `get_ticker_history` will return the data w/o duplications and only starting from `since` time. So there could be the duplication only if CCXT library has a bug in it(it could be, like this one https://github.com/ccxt/ccxt/issues/379, but looks like they all were fixed;) ).

I can revert this code if you want. But it will be like a double-cleck of CCXT response.

Reply:
Makes sense to me (also, i did miss the connection to line 198 - which sets since to the last point downloaded previously).